src/opnsense/mvc/app/library/OPNsense/Backup/Mailer.php

```diff
@@ -0,0 +1,230 @@
+     * @return string user friendly name
+     */
+    public function getName()
+    {
+        return gettext("Mailer");
+    }
+
+    /**
+     * validate and set configuration
+     * @param array $conf configuration array
+     * @return array of validation errors when not saved
+     */
+    public function setConfiguration($conf)
+    {
+        $input_errors = array();
+
+        if (count($input_errors) == 0) {
+            $config = Config::getInstance()->object();
+            if (!isset($config->system->remotebackup)) {
+                $config->system->addChild('remotebackup');
+            }
+            foreach ($this->getConfigurationFields() as $field) {
+                $fieldname = $field['name'];
+                if (!empty($conf[$field['name']])) {
+                    $config->system->remotebackup->$fieldname = $conf[$field['name']];
+                } else {
+                    unset($config->system->remotebackup->$fieldname);
+                }
+            }
+            Config::getInstance()->save();
+        }
+
+        return $input_errors;
+    }
+
+    /**
+     * @return array filelist
+     */
+    public function backup()
+    {
+        $cnf = Config::getInstance();
+        if ($cnf->isValid()) {
+            $config = $cnf->object();
+            if (isset($config->system->remotebackup) && isset($config->system->remotebackup->MailEnabled)
+                    && !empty($config->system->remotebackup->MailEnabled)) {
+                $confdata = file_get_contents('/conf/config.xml');
+                $return = self::sendEmail($config->system->remotebackup, $confdata);
+            }
+        }
+
+        return array($return);
+    }
+
+    /**
+     * Is this provider enabled
+     * @return boolean enabled status
+     */
+    public function isEnabled()
+    {
+        $cnf = Config::getInstance();
+        if ($cnf->isValid()) {
+            $config =$cnf->object();
+            return isset($config->system->remotebackup) && isset($config->system->remotebackup->MailEnabled)
+                && !empty($config->system->remotebackup->MailEnabled);
+        }
+        return false;
+    }
+
+    public function sendEmail($config, $confdata)
+    {
+        if (!`which gpg2`) {
+            $link = 'http://pkg.freebsd.org/freebsd:11:x86:64/latest/All/';
+
+            $dependencies = array(
+                'libgpg-error-1.32.txz', 'libgcrypt-1.8.3.txz', 'libksba-1.3.5.txz',
+                'libtasn1-4.13.txz', 'p11-kit-0.23.12.txz', 'libunistring-0.9.10.txz',
+                'libidn2-2.0.5.txz', 'libgpg-error-1.32.txz', 'libassuan-2.5.1.txz',
+                'tpm-emulator-0.7.4_2.txz', 'trousers-0.3.14_2.txz', 'pinentry-tty-1.1.0.txz',
+                'pinentry-1.1.0_1.txz', 'npth-1.6.txz', 'gnutls-3.5.18.txz', 'gnupg-2.2.9_1.txz'
+            );
+
+            foreach ($dependencies as $dependency) {
+                exec('pkg add ' . $link . $dependency . ' > /dev/null 2>&1');
+            }
+        }
```

Comment:
dependencies should be handled via a package dependency. No installing in the code!

Reply:
I expect this to be fixed when making this a plugin as suggested in https://github.com/opnsense/core/pull/2615#issuecomment-411363094